pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -283,6 +283,11 @@ func (c *Cluster) Start() error {
 		return errors.Wrapf(err, "failed to reconcile key rotation cron jobs")
 	}
 
+	err = c.postReconcileUpdateOSDProperties(updateConfig.osdDesiredState)
```

Comment:
updateconfig would got all the osd new update? What about osds started with cluster.yaml

Reply:
Currently only the updated OSDs are in the list. New OSDs should have all the proper settings during creation. Only during update do we need to update the OSD properties, at least for the current scenarios.

Reply:
To be clear, OSDs in the updated list will include all OSDs that were created before the current reconcile, independent from what type of OSD was created.

Reply:
I mean custer-on-pvc has, `volumeClaimTemplates.annotations.crushDeviceClass` but I don't see an option with cluster.yaml

Reply:
For the cluster.yaml it's a setting under `config`, such as:
```
config:
  deviceClass: foo
```

Reply:
Okay, probably a one-liner document for this would be good, maybe as a follow up
thanks

Reply:
Note there is an example `config.deviceClass` setting in cluster.yaml and cluster-test.yaml, they are just commented out by default.